src/components/Api.js

```diff
@@ -0,0 +1,91 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    this._authorization = headers.authorization;
+    this._contentType = headers.contentType;
+  }
+
+  // GENERAL PROMISES //
+  _handleServerResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  // CARDS //
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, { headers: this._headers })
+      .then(this._handleServerResponse)
+      .catch((err) => console.error(`Api error getting cards, ${err}`));
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.